viral/offline_sequence_detection.py

```
@@ -0,0 +1,857 @@
+    Essentially, this is a Python implementation of https://github.com/losonczylab/Grosmark_NatNeuro_2021/blob/main/calcWeightedLinearCorr.m
+    """
+    n_time, n_pos = posterior_probability_matrix.shape
+
+    time_coords = np.repeat(np.arange(n_time), n_pos)
+    pos_coords = np.tile(np.arange(n_pos), n_time)
+
+    xy = np.column_stack((time_coords, pos_coords))
+
+    w = posterior_probability_matrix.flatten()
+    if np.isnan(np.any(w)):
+        raise ValueError("Encountered NaN in posterior probability matrix")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)